engine/src/main/java/org/destinationsol/game/SolCam.java

```diff
@@ -61,31 +65,30 @@ public Matrix4 getMtx() {
     public void update(SolGame game) {
         float life = 0;
 
-        SolShip hero = game.getHero();
+        Hero hero = game.getHero();
         float ts = game.getTimeStep();
-        if (hero == null) {
-            StarPort.Transcendent trans = game.getTranscendentHero();
-            if (trans == null) {
-                if (DebugOptions.DIRECT_CAM_CONTROL) {
-                    applyInput(game);
-                }
-            } else {
-                myPos.set(trans.getPosition());
-            }
+        if (hero.isDead() || DIRECT_CAM_CONTROL) {
+            applyInput(game);
+            life = hero.getLife();
```

Comment:
Later in the code, screen is shaken, if life is less than the previous amount was. And since life is initially set to `0`, every time directcamcontrol is pressed, screen would shake, since life would be `0` and that would be less than the previous amount was. I however *could* extract the `life = ...`